File: events.php
<?php

class Event {
    public $id;
    public $month;
    public $monthNameAr;
    public $monthNameEn;
    public $day;
    public $yearOfOccurrence;
    public $name;
    public $nameEn;
    public $description;
    public $importance;
    public $typeAr;
    public $typeEn;

    const ID_KEY = 'id';
    const MONTH_KEY = 'month';
    const MONTH_NAME_AR_KEY = 'month_name_ar';
    const MONTH_NAME_EN_KEY = 'month_name_en';
    const DAY_KEY = 'day';
    const YEAR_OF_OCCURRENCE_KEY = 'year_of_occurrence';
    const NAME_KEY = 'name';
    const NAME_EN_KEY = 'name_en';
    const DESCRIPTION_KEY = 'description';
    const IMPORTANCE_KEY = 'importance';
    const TYPE_AR_KEY = 'type_ar';
    const TYPE_EN_KEY = 'type_en';

    public function __construct($json) {
        $this->id = $json[self::ID_KEY] ?? null;
        $this->month = $json[self::MONTH_KEY] ?? null;
        $this->monthNameAr = $json[self::MONTH_NAME_AR_KEY] ?? null;
        $this->monthNameEn = $json[self::MONTH_NAME_EN_KEY] ?? null;
        $this->day = $json[self::DAY_KEY] ?? null;
        $this->yearOfOccurrence = $json[self::YEAR_OF_OCCURRENCE_KEY] ?? null;
        $this->name = $json[self::NAME_KEY] ?? null;
        $this->nameEn = $json[self::NAME_EN_KEY] ?? null;
        $this->description = $json[self::DESCRIPTION_KEY] ?? null;
        $this->importance = $json[self::IMPORTANCE_KEY] ?? null;
        $this->typeAr = $json[self::TYPE_AR_KEY] ?? null;
        $this->typeEn = $json[self::TYPE_EN_KEY] ?? null;
    }
}

class Events {
    public $events = [];

    public function initEvents() {
        try {
            $content = file_get_contents('events.json');
            if ($content === false) {
                throw new Exception("Unable to read file");
            }
            $json = json_decode($content, true);
            if ($json === null) {
                throw new Exception("Error parsing JSON");
            }
            $list = $json['islamic_events'];
            foreach ($list as $item) {
                $this->events[] = new Event($item);
            }
        } catch (Exception $e) {
            echo 'Error reading or parsing events.json: ',  $e->getMessage(), "\n";
        }
    }
}

if (basename(__FILE__) == basename($_SERVER["SCRIPT_FILENAME"])) {
    $events = new Events();
    $events->initEvents();
    foreach ($events->events as $e) {
        echo "يوم {$e->day} {$e->monthNameAr} - {$e->name}\n";
    }
}

?>

File: events.py
import json
import os

class Event:
    id_key = 'id'
    month_key = 'month'
    month_name_ar_key = 'month_name_ar'
    month_name_en_key = 'month_name_en'
    day_key = 'day'
    year_of_occurrence_key = 'year_of_occurrence'
    name_key = 'name'
    name_en_key = 'name_en'
    description_key = 'description'
    importance_key = 'importance'
    type_ar_key = 'type_ar'
    type_en_key = 'type_en'

    def __init__(self, data):
        self.id = data.get(self.id_key)
        self.month = data.get(self.month_key)
        self.month_name_ar = data.get(self.month_name_ar_key)
        self.month_name_en = data.get(self.month_name_en_key)
        self.day = data.get(self.day_key)
        self.year_of_occurrence = data.get(self.year_of_occurrence_key)
        self.name = data.get(self.name_key)
        self.name_en = data.get(self.name_en_key)
        self.description = data.get(self.description_key)
        self.importance = data.get(self.importance_key)
        self.type_ar = data.get(self.type_ar_key)
        self.type_en = data.get(self.type_en_key)

class Events:
    def __init__(self):
        self.events = []

    def init_events(self):
        try:
            with open('events.json', 'r', encoding='utf-8') as f:
                content = f.read()
                data = json.loads(content)
                event_list = data['islamic_events']
                self.events = [Event(e) for e in event_list]
        except Exception as e:
            print(f"Error reading or parsing events.json: {e}")

if __name__ == "__main__":
    events = Events()
    events.init_events()
    for e in events.events:
        print(f"يوم {e.day} {e.month_name_ar} - {e.name}")
